dropshot/src/server.rs

```diff
@@ -1019,38 +1058,10 @@ impl<C: ServerContext> ServerConnectionHandler<C> {
     fn new(server: Arc<DropshotState<C>>) -> Self {
         ServerConnectionHandler { server }
     }
-}
-
-impl<T: ServerContext> Service<&AddrStream> for ServerConnectionHandler<T> {
-    // Recall that a Service in this context is just something that takes a
-    // request (which could be anything) and produces a response (which could be
-    // anything).  This being a connection handler, the request type is an
-    // AddrStream (which wraps a TCP connection) and the response type is
-    // another Service: one that accepts HTTP requests and produces HTTP
-    // responses.
-    type Response = ServerRequestHandler<T>;
-    type Error = GenericError;
-    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
-
-    fn poll_ready(
-        &mut self,
-        _cx: &mut Context<'_>,
-    ) -> Poll<Result<(), Self::Error>> {
-        // TODO is this right?
-        Poll::Ready(Ok(()))
-    }
 
-    fn call(&mut self, conn: &AddrStream) -> Self::Future {
-        // We're given a borrowed reference to the AddrStream, but our interface
-        // is async (which is good, so that we can support time-consuming
-        // operations as part of receiving requests).  To avoid having to ensure
-        // that conn's lifetime exceeds that of this async operation, we simply
-        // copy the only useful information out of the conn: the SocketAddr.  We
-        // may want to create our own connection type to encapsulate the socket
-        // address and any other per-connection state that we want to keep.
-        let server = Arc::clone(&self.server);
-        let remote_addr = conn.remote_addr();
-        Box::pin(http_connection_handle(server, remote_addr))
+    fn make_svc(&self, remote_addr: SocketAddr) -> ServerRequestHandler<C> {
+        info!(self.server.log, "accepted connection"; "remote_addr" => %remote_addr);
```

Comment:
This raised my eyebrow a bit and I think I figured out what happened.

Before this PR, both the InnerHttpServerStarter and InnerHttpsServerStarter invoked `http_connection_handle()` to log this message and produce the `ServerRequestHandler`.  This was done in the `impl Service<...> for ServerConnectionHandler` (where the type parameter was `&AddrStream` for the HTTP case and `TlsConn` for the TLS case).  This PR removes `impl Service<&AddrStream>`, but not the `impl Service<&TlsConn>`.  However, the `impl Service<&TlsConn>` is unused.  I removed it, which exposed that `http_connection_handle` is also unused now.  But this `make_svc` function I think is identical to what that function was.

So I think what I'd suggest is:
- remove the `impl Service<&TlsConn> for ServerConnectionHandler`
- remove `make_svc()` and have its callers call `http_connection_handle` instead?  Alternatively, rename this function to `make_http_request_handler` (which I find a lot more descriptive).  But the reason I had this as a naked top-level function is so that it would be obvious in the control flow at what point a new connection has been received so that if needed to put other logic there, we could do it there.